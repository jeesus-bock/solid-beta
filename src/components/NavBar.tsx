import { Link } from '@solidjs/router';
import style from './NavBar.module.css';
export const NavBar = () => {
  return (
    <nav class={style.nav}>
      <Link href='/'>Home</Link>
      <Link href='/about'>About</Link>
      <Link href='/magicks'>Magicks</Link>
    </nav>
  );
};
