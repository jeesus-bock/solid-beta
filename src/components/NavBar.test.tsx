import { describe, expect, test } from 'vitest';
import { render } from 'solid-testing-library';
import { NavBar } from './NavBar';
import { Router } from '@solidjs/router';

/* NavBar requires the Router, so the renders() have to use it.
 */
describe('<NavBar />', () => {
  test('Renders', async () => {
    const { container, unmount } = await render(() => (
      <Router>
        <NavBar />
      </Router>
    ));
    expect(container).toMatchSnapshot();
    unmount();
  });
  test('Renders nav link texts', async () => {
    const { container, unmount } = await render(() => (
      <Router>
        <NavBar />
      </Router>
    ));
    const childTexts = [...(container.firstChild?.childNodes || [])].map((cn) => cn.textContent);
    expect(childTexts).toEqual(['Home', 'About', 'Magicks']);
    unmount();
  });
});
