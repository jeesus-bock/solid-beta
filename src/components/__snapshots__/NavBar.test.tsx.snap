// Vitest Snapshot v1

exports[`<NavBar /> > Renders 1`] = `
<div>
  <nav
    class="_nav_1sufr_1"
  >
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/about"
    >
      About
    </a>
    <a
      href="/magicks"
    >
      Magicks
    </a>
  </nav>
</div>
`;
